modules/post/windows/gather/credentials/securecrt.rb

```diff
@@ -62,8 +62,13 @@ def try_encode_file(data)
   def enum_session_file(path)
     config_ini = []
     tbl = []
-    print_status("Searching for session files in #{path}")
-    config_ini += session.fs.file.search(path, '*.ini')
+    begin
+      print_status("Searching for session files in #{path}")
+      config_ini += session.fs.file.search(path, '*.ini')
+      fail_with(Failure::BadConfig, "Couldn't find any session files at #{path}") if config_ini.to_s.empty?
```

Comment:
I could be wrong, but I think @dwelch-r7 was originally suggesting:
```suggestion
      fail_with(Failure::BadConfig, "Couldn't find any session files at #{path}") if config_ini.empty?
```

Converting to a string isn't necessary as `config_ini` is being initialised to an empty array - and therefore there won't be any nil scenarios to be caught out by 👍 

That aside, this guard clause is now always true as an empty array to_s is the string `"[]"`:

```ruby
2.6.5 :005 > [].to_s
 => "[]" 
```

A quick example showing the unintended behavior:

```
2.6.5 :006 > [].to_s.empty?
 => false 
2.6.5 :007 > ["a", "b", "c"].to_s.empty?
 => false
```

Compared to the expected output:

```ruby
2.6.5 :008 > [].empty?
 => true 
2.6.5 :009 >  ["a", "b", "c"].empty?
 => false 
```

Reply:
Oh woops I honestly thought it would return a blank string. Thanks for checking that! Also agree on the proposed change given that this is just going to be an array, it was more to future proof the code in case the type did change but I suppose that wasn't entirely needed and given this behavior its really not desired 😄 

Let me change that over quick.

Reply:
Tested and all looks good other than this, once you get that switched over I'd be happy to land :+1:

Reply:
@dwelch-r7 Woops I said I was going to fix this and then I didn't do that 😅 Fixing now.